app/javascript/stimulus/search_controller.js

```diff
@@ -0,0 +1,37 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = [ "input", "clear" ]
+
+  inputTargetConnected() {
+    this.setSearchClearIcon()
+  }
```

Comment:
I needed to change this from connect() to inputTargetConnected() but according to stimulus docs that should not make a difference. I suspect we may be working around a stimulus bug. I filed a ticket here: https://github.com/hotwired/turbo/issues/1351